src/Illuminate/Database/Eloquent/Relations/MorphTo.php

```diff
@@ -123,7 +123,9 @@ protected function getResultsByType($type)
                                 (array) ($this->morphableEagerLoads[get_class($instance)] ?? [])
                             ));
 
-        return $query->whereIn(
+        $whereIn = $this->whereInMethod($this->parent, $ownerKey);
```

Comment:
We need to check the related model, not the parent:

```suggestion
        $whereIn = $this->whereInMethod($instance, $ownerKey);
```

Reply:
You are right, I tried it and it make more sense. Thanks